tests/procfs/procfs.c

```diff
@@ -93,14 +94,37 @@ int test_readonly()
 
 int test_maps()
 {
-    int fd;
-    char buf[1024];
-
-    fd = open("/proc/self/maps", O_RDONLY);
-    assert(fd > 0);
-    assert(read(fd, buf, sizeof(buf)));
-
-    printf("%s\n", buf);
+    char buf[16 * 1024];
+    void* addr;
+
+    {
+        const int prot = PROT_READ | PROT_WRITE | PROT_EXEC;
+        const int flags = MAP_PRIVATE | MAP_ANONYMOUS;
+        const size_t length = 4096 * 1024;
+        if ((addr = mmap(NULL, length, prot, flags, -1, 0)) == MAP_FAILED)
+            assert(0);
+    }
+
+    struct stat statbuf;
+    assert(stat("/datafile", &statbuf) == 0);
+    int fd = open("/datafile", O_RDWR);
+    assert(fd >= 0);
+
+    {
+        const int prot = PROT_READ | PROT_WRITE;
+        const int flags = MAP_PRIVATE | MAP_FIXED;
+        const size_t length = statbuf.st_size / 2;
+        if (mmap(addr, length, prot, flags, fd, 8192) == MAP_FAILED)
+            assert(0);
+    }
+
+    {
+        int fd = open("/proc/self/maps", O_RDONLY);
+        assert(fd > 0);
+        assert(read(fd, buf, sizeof(buf)));
+        printf("%s\n", buf);
```

Comment:
Can we assert that datafile was found in the mappings.

Reply:
Good idea. I will try.

Reply:
I added a more elaborate test to check for existence of ``/datafile`` and also to do a partial unmap and check that there are now two entries.